input/reproSchemaScript/reprotofhirjson.py

```diff
@@ -0,0 +1,282 @@
+'''
+script to convert repro scheme to json for fhir questionnaire
+Not perfect conversion
+#example: "python reprotofhirjson.py <reproschema_folder> valueset"
+must be in the file structure specified in:
+https://github.com/ReproNim/reproschema-library/tree/43e7afab312596708c0ad4dfd45b69c8904088ae/activities
+
+'''
+import json
+import sys
+import os
+import logging
+from dotenv import load_dotenv
+
+load_dotenv()
+CODESYSTEM_URI = os.getenv('CODESYSTEM_URI')
+VALUESET_URI = os.getenv('VALUESET_URI')
+QUESTIONNAIRE_URI = os.getenv('QUESTIONNAIRE_URI')
+
+codeSystem_dict = dict()
+
+fhir_questionnaire = dict()
+
+fhir_valuesets = []
+fhir_codesystems = []
+
+
+
+def generate_codeSystem(options_json, linkId,questionnaire):
+   # default headers for codesystem
+    codeSystem = dict()
+    id = questionnaire + linkId
+    id = id.replace("_","-")
+    id = id.lower()
+
+    codeSystem["resourceType"] = "CodeSystem"
+    codeSystem["id"] = id
+    codeSystem["text"] = {
+    "status" : "generated",
+    "div" : "<div xmlns=\"http://www.w3.org/1999/xhtml\">Placeholder</div>"}
+
+    codeSystem["url"] = str(CODESYSTEM_URI) + id
+    codeSystem["version"] = "1.4.0"
+    codeSystem["name"] = (id).capitalize().replace("_","")
+    codeSystem["title"] = id
+    codeSystem["status"] = "active"
+    codeSystem["date"] = "2023-11-20T11:33:15-05:00"
+    codeSystem["publisher"] = "KinD Lab"
+    codeSystem["contact"] = [
+        {
+        "name" : "KinD Lab",
+        "telecom" : [
+            {
+            "system" : "url",
+            "value" : "http://fhir.kindlab.sickkids.ca"
+            }
+        ]
+        }
+    ]
+
+    codeSystem["description"] = id
+    codeSystem["caseSensitive"] = True
+    codeSystem["content"] = "complete"
+    codeSystem["count"] = len(options_json["choices"])
+    codeSystem["concept"] = []
+    
+    options = []
+    # loops through the options and appends them to the codesystem
```

Comment:
In general this is the type of comment you want to avoid making, because it's *too* literal.

It tells me that you loop through the options and then append them. Then I read the code and the code is `for j in options_json ...` (a loop) followed by `options.append(choice)` (an append). So in fact the comment hasn't given me much added info over the code itself. Instead try to write a comment that explains *why* you are doing this. My understanding is you are collecting each choice in the reproSchema input and converting them into a code/display to be added to the code system in FHIR. So the ideal comment would say: (1) what does reproSchema store in "choices" (e.g. what is their data structure), and (2) how is it equivalently stored in FHIR (what's your target structure), and (3) how you are going to go from A->B